src/libraries/System.IO.Compression/src/System/IO/Compression/ZLibStream.cs

```diff
@@ -0,0 +1,293 @@
+        /// <param name="compressionLevel">One of the enumeration values that indicates whether to emphasize speed or compression efficiency when compressing the stream.</param>
+        /// <param name="leaveOpen"><c>true</c> to leave the stream object open after disposing the <see cref="ZLibStream"/> object; otherwise, <c>false</c>.</param>
+        public ZLibStream(Stream stream, CompressionLevel compressionLevel, bool leaveOpen)
+        {
+            _deflateStream = new DeflateStream(stream, compressionLevel, leaveOpen, ZLibNative.ZLib_DefaultWindowBits);
+        }
+
+        /// <summary>Gets a value indicating whether the stream supports reading.</summary>
+        public override bool CanRead => _deflateStream?.CanRead ?? false;
+
+        /// <summary>Gets a value indicating whether the stream supports writing.</summary>
+        public override bool CanWrite => _deflateStream?.CanWrite ?? false;
+
+        /// <summary>Gets a value indicating whether the stream supports seeking.</summary>
+        public override bool CanSeek => _deflateStream?.CanSeek ?? false;
+
+        /// <summary>This property is not supported and always throws a <see cref="NotSupportedException"/>.</summary>
+        public override long Length => throw new NotSupportedException();
+
+        /// <summary>This property is not supported and always throws a <see cref="NotSupportedException"/>.</summary>
+        public override long Position
+        {
+            get => throw new NotSupportedException();
+            set => throw new NotSupportedException();
+        }
+
+        /// <summary>Flushes the internal buffers.</summary>
+        public override void Flush()
+        {
+            ThrowIfClosed();
+            _deflateStream.Flush();
+        }
+
+        /// <summary>Asynchronously clears all buffers for this stream, causes any buffered data to be written to the underlying device, and monitors cancellation requests.</summary>
+        /// <param name="cancellationToken">The token to monitor for cancellation requests.</param>
+        /// <returns>A task that represents the asynchronous flush operation.</returns>
+        public override Task FlushAsync(CancellationToken cancellationToken)
+        {
+            ThrowIfClosed();
+            return _deflateStream.FlushAsync(cancellationToken);
+        }
+
+        /// <summary>This method is not supported and always throws a <see cref="NotSupportedException"/>.</summary>
+        public override long Seek(long offset, SeekOrigin origin) => throw new NotSupportedException();
+
+        /// <summary>This method is not supported and always throws a <see cref="NotSupportedException"/>.</summary>
+        public override void SetLength(long value) => throw new NotSupportedException();
+
+        /// <summary>Reads a byte from the stream and advances the position within the stream by one byte, or returns -1 if at the end of the stream.</summary>
+        /// <returns></returns>
```

Comment:
Since this returns what `_deflateStream.ReadByte()` returns, we can copy the original returns string:
```suggestion
        /// <returns>The unsigned byte cast to an <see cref="int" />, or -1 if at the end of the stream.</returns>
```
Original: https://docs.microsoft.com/en-us/dotnet/api/system.io.compression.deflatestream.readbyte?view=netcore-3.1